noir-projects/aztec-nr/aztec/src/state_vars/private_set.nr

```diff
@@ -44,35 +44,41 @@ impl<Note> PrivateSet<Note> {
     // DEPRECATED
     fn assert_contains_and_remove(_self: Self, _note: &mut Note, _nonce: Field) {
         assert(
-            false, "`assert_contains_and_remove` has been deprecated. Please call PXE.addNote() to add a note to the database. Then use PrivateSet.get_notes() and PrivateSet.remove() in your contract to verify and remove a note."
+            false, "`assert_contains_and_remove` has been deprecated. Please call PXE.addNote() to add a note to the database. Then use PrivateSet.get_notes() in your contract to verify and remove a note."
         );
     }
 
     // DEPRECATED
     fn assert_contains_and_remove_publicly_created(_self: Self, _note: &mut Note) {
         assert(
-            false, "`assert_contains_and_remove_publicly_created` has been deprecated. Please call PXE.addNote() to add a note to the database. Then use PrivateSet.get_notes() and PrivateSet.remove() in your contract to verify and remove a note."
+            false, "`assert_contains_and_remove_publicly_created` has been deprecated. Please call PXE.addNote() to add a note to the database. Then use PrivateSet.get_notes() in your contract to verify and remove a note."
         );
     }
 
-    // docs:start:remove
-    pub fn remove<N>(self, note: Note) where Note: NoteInterface<N> {
-        let context = self.context.private.unwrap();
-        let note_hash = compute_note_hash_for_read_request(note);
-        let has_been_read = context.note_hash_read_requests.any(|r: ReadRequest| r.value == note_hash);
-        assert(has_been_read, "Can only remove a note that has been read from the set.");
-
-        destroy_note(context, note);
-    }
-    // docs:end:remove
-
     // docs:start:get_notes
     pub fn get_notes<N, FILTER_ARGS>(
         self,
         options: NoteGetterOptions<Note, N, FILTER_ARGS>
     ) -> [Option<Note>; MAX_NOTE_HASH_READ_REQUESTS_PER_CALL] where Note: NoteInterface<N> {
+        let context = self.context.private.unwrap();
+
         let storage_slot = self.storage_slot;
-        let opt_notes = get_notes(self.context.private.unwrap(), storage_slot, options);
+        let opt_notes = get_notes(context, storage_slot, options);
```

Comment:
My brain must be half asleep. Got semi confused with the get notes destroying notes in here, but was just that get_notes call get_notes 💀